src/lib/TextExtraction.js

```diff
@@ -31,7 +31,7 @@ class TextExtraction {
         let textLeft = parsedText.children;
 
         while (textLeft) {
-          let matches = pattern.pattern.exec(textLeft);
+          let matches = new RegExp(pattern.pattern).exec(textLeft);
```

Comment:
Would be _slightly_ more efficient to only clone it once outside the while loop...